hngprojects/hng_boilerplate_nestjs
# feat: two-factor authentication - verify

## What does this Pull Request Do?
This PR is about verifying a two factor authentication code for additional layer of security.

## How should this be manually tested?
As an authenticated user, send a POST request to `/api/v1/auth/2fa/verify` with a request body as below:
```bash
{
    "totp_code": "123456"
}
```
## Checklist of what you did
- [x] Implemented endpoint to verify the TOTP for 2fa authentication.
- [x] Generate and store backup codes
- [x] Returned success message with the right response payload.
- [x] Added unit tests for the auth service

Issue Link: [Issue 266](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/266)
